lisa/sut_orchestrator/azure/features.py

```diff
@@ -2477,27 +2477,29 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
                     for zone in params.availability_zones
                     if zone in settings.availability_zones
                 ]
-                assert params.availability_zones, (
-                    "Invalid zones provided. "
-                    "This SKU in this location supports zones: "
-                    f"{settings.availability_zones}. "
-                )
+                if not params.availability_zones:
+                    raise SkippedException(
+                        "Invalid zones provided. "
+                        "This SKU in this location supports zones: "
+                        f"{settings.availability_zones}. "
+                    )
             elif settings.availability_zones:
                 params.availability_zones = [settings.availability_zones.items[0]]
 
             assert params.availability_type in [
                 type.value for type in AvailabilityType
             ], ("Not a valid Availability Type: " f"{params.availability_type}")
 
-            assert (
+            if not (
                 AvailabilityType(params.availability_type) in settings.availability_type
-            ), (
-                f"Availability Type "
-                f"'{params.availability_type}' "
-                "is not supported in the current configuration. Please select one of "
-                f"{[type.value for type in settings.availability_type.items]}. "
-                "Or consider changing the disk type or location."
-            )
+            ):
+                raise SkippedException(
+                    f"Availability Type "
+                    f"'{params.availability_type}' "
+                    "is not supported in the current configuration. Please select one of "
+                    f"{[type.value for type in settings.availability_type.items]}. "
+                    "Or consider changing the disk type or location."
+                )
```

Comment:
Do you think we should add this statement "consider changing the disk type or location.", in error log only if is_ultra_disk is true?
I feel it would be misleading otherwise.

Reply:
I don't think it is an issue. The only case you hit this error message without having ultra disk enabled is if you explicitly state zone redundancy in the runbook. Zone redundancy is not really used outside of ultra disk deployment.

Did you have a specific scenario in mind where this message would be exceedingly misleading?

Reply:
For the error message, it's read by human. It doesn't need to be accurate or an exact result. So, you can try to provide most or all possible reasons to help troubleshooting.

Reply:
I plan to add AvailabilityType.NoRedudancy as a test case requirement for Hibernation.
If a Runbook runs Hibernation, but has availability_type set to AvailabilitySet, it would raise this SkippedException, but would have nothing to do with the disk

Reply:
@kamalca @adityagesh let know your discussion results, then I can proceed this PR, thanks.

Reply:
I have made the language more clear. @LiliDeng you can proceed, thank you.